### ISSUE TITLE
Crash on channel view

### ISSUE THREAD

**Issue by notxenonbox** [uowuo/abaddon#26]
If I click on a DM or a guild channel abaddon crashes.

I am on arch linux.
I installed the `abaddon-git` AUR package and my system is up to date

here is the output
```
neo@fukushima ~
λ abaddon
sending {"d":{"capabilities":61,"client_state":{"guild_hashes":{},"highest_last_message_id":"0","read_state_version":0,"user_guild_settings_version":-1},"compress":false,"presence":{"activities":[],"afk":false,"since":0,"status":"online"},"properties":{"browser":"","browser_user_agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36","browser_version":"67.0.3396.87","client_build_number":82826,"client_event_source":null,"device":"Chrome","os":"Windows","os_version":"10","referrer":"","referrer_current":"","referring_domain":"","referring_domain_current":"","release_channel":"stable","system_locale":"en-US"},"token":"[REDACTED]"},"op":2}
sending {"d":null,"op":1}
Unknown event SESSIONS_REPLACE
GET /channels/662673667809083452/messages?limit=50
sending {"d":{"activities":true,"channels":{"662673667809083452":[[0,99],[100,199]]},"guild_id":"638020853107523604","typing":true},"op":14}
/usr/bin/abaddon: line 3:  7808 Segmentation fault      (core dumped) ./abaddon
neo@fukushima ~ 6s
λ abaddon
sending {"d":null,"op":1}
sending {"d":{"capabilities":61,"client_state":{"guild_hashes":{},"highest_last_message_id":"0","read_state_version":0,"user_guild_settings_version":-1},"compress":false,"presence":{"activities":[],"afk":false,"since":0,"status":"online"},"properties":{"browser":"","browser_user_agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36","browser_version":"67.0.3396.87","client_build_number":82826,"client_event_source":null,"device":"Chrome","os":"Windows","os_version":"10","referrer":"","referrer_current":"","referring_domain":"","referring_domain_current":"","release_channel":"stable","system_locale":"en-US"},"token":"[REDACTED]"},"op":2}
/usr/include/c++/10.2.0/optional:448: constexpr const _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() const [with _Tp = Snowflake; _Dp = std::_Optional_base<Snowflake, true, true>]: Assertion 'this->_M_is_engaged()' failed.
/usr/bin/abaddon: line 3:  7843 Aborted                 (core dumped) ./abaddon
```

The first one is the guild channel, second is the DM.
If there is a need for more information, just ask.

_I hope I didn't just leak my token_

**Comment by javsanpar** [comment]
I will try to reproduce the crash myself.

**Comment by javsanpar** [comment]
I can not reproduce the crash with the last version.

I just bumped the package version. If you can, update the package and try again.

**Comment by ouwou** [comment]
it looks like i made a wrong assumption about what the API could return somewhere. can you email me the crash dumps `shtorf [at] protonmail.com`? also does this happen for every dm or channel or just a couple, and does it crash as soon as you click or does it crash after a small delay (i.e. after the HTTP request for messages completes)?

**Comment by notxenonbox** [comment]
Sorry for the late response, it happens every DM (tried two so far) and it happens on every (the three I tried) text channel. It crashes with a bit of a delay. I sent the email with all my crash dumps. I will try the bumped version later today

**Comment by ouwou** [comment]
think you can get a backtrace yourself? im having some trouble trying to do it

**Comment by ouwou** [comment]
@notxenonbox can you get the backtrace? if i dont hear back for a couple more days ill close this but you can reopen if that happens

**Comment by notxenonbox** [comment]
I'm sorry for disappearing, but life got in the way, I can't get the backtrace, I'm sorry. I will reopen the issue if I come back to try running abaddon again.

**Comment by ouwou** [comment]
i think i ran into this and fixed it in a4aab9bc164950e543525c0d80668956855d187f
